### PR TITLE
CMake changes

### PR DESCRIPTION
**Pull request by connojd**
Introduce a new target-centric cmake design

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -44,6 +44,7 @@ ignore:
   - "bfintrinsics/**"
   - "bfm/ioctl/arch/**"
   - "bfvmm/include/test/**"
+  - "bfdriver/**"
```
Comment (rianquinn) [inline .codecov.yml]: why was this added? I'd like to work at reducing these. Not adding to them.

File: .doxygen.txt [hunk 2]
```diff
@@ -43,7 +43,7 @@ PROJECT_BRIEF = "lightweight, extendable hypervisor written in C++ with support
 # spaces.
 # Note: If this tag is empty the current directory is searched.
 
-INPUT = bfdriver/include bfelf_loader/include bfm/include bfsdk/include bfvmm/include
```
Comment (rianquinn) [inline .doxygen.txt]: why was this removed?

File: .travis.yml [hunk 3]
```diff
@@ -275,127 +291,3 @@ matrix:
...
-          - clang-6.0
-          - nasm
-    env:
-      - TEST="GCC 8"
-    script:
-      - cmake -DCONFIG=travis_shared ..
-      - make -j3
-
-  #
-  # ld.gold
```
Comment (rianquinn) [inline .travis.yml]: why are all of these tests removed?

File: .doxygen.txt (hunk 2 above)
Comment (connojd) [reply .doxygen.txt]: I moved the elf loader to bfsdk

File: .codecov.yml (hunk 1 above)
Comment (connojd) [reply .codecov.yml]: The driver tests arent being built right now because we dropped shared library support

**Comment by JaredWright** [comment]
I think this is HUGE improvement, great work. So far, I've run into one issue: dependencies between top level projects seem to be missing, breaking parallel builds on my environment. Specifically, the bfvmm subproject seems to be missing direct (or transitive) dependencies on newlib, binutils, libcxx, libcxxabi, bfintrinsics, bfsdk. Are you having issues with parallel builds on your environment?

Also, it looks like the next ubuntu release (19.04) comes with cmake 3.13 (woo hoo!) We should keep it on our radar to update the README next month to make that our minimum supported versions, since this PR moves to cmake > 3.13

**Comment by JaredWright** [comment]
Changing top-level CMakeLists.txt:139 from ```add_subproject(bfvmm vmm DEPENDS bfunwind)``` to ``` add_subproject(bfvmm vmm DEPENDS bfunwind bfintrinsics) ``` fixed the dependency issues on my environment
